src/lib.rs

```diff
@@ -181,7 +181,7 @@ pub struct Type {
 
 /// Enum with additional information, depending on the kind of type.
 // Clone is used only for error reporting and is not intended for end users
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq, Hash, Eq)]
```

Comment:
I think with the changes to `LocalType` that I suggested, these derives are no longer going to be needed